### PR TITLE
Older devices are sending "gpc": 1 and this can't marshal.

### PR DESCRIPTION
**Pull request by scr-oath**
- See https://github.com/prebid/Prebid.js/issues/12435

### CODE REVIEW

File: openrtb_ext/request_wrapper_test.go
Comment (SyntaxNode) [inline openrtb_ext/request_wrapper_test.go:2348]: Nitpick: The Go test runner replaces spaces in test names with an underscore. To make it easier to search for tests by name, please do not use spaces. Consider renaming to  `valid_gpc_json_"1"` and `valid_gpc_json_1`,

File: openrtb_ext/request_wrapper_test.go
Comment (SyntaxNode) [inline openrtb_ext/request_wrapper_test.go:1]: There is a pre-existing issue in this the `TestRegExtUnmarshal` test which never asserts the GPC field. Could you please add an `assert.Equal(t, tt.expectGPC, tt.regExt.gpc)` within the `t.Run` func?

File: openrtb_ext/request_wrapper_test.go
Comment (scr-oath) [reply openrtb_ext/request_wrapper_test.go]: DONE

File: openrtb_ext/request_wrapper_test.go
Comment (scr-oath) [reply openrtb_ext/request_wrapper_test.go:1]: DONE

**Review by SyntaxNode — APPROVED** [review]

File: openrtb_ext/request_wrapper.go
Comment (scr-oath) [reply openrtb_ext/request_wrapper.go:1285]: I saw that gjson was already being used, and I know that it's much more lenient. Are you suggesting that we roll this fix into the jsonutil library somehow? like a `ReadString` method or something to that effect (that could be the thing using gjson)

- 51 Degrees already "broke that seal" #3893

Comment (scr-oath) [reply openrtb_ext/request_wrapper.go:1285]: I moved this into a new jsonutil function `ParseIntoString` LMK if you are ok with that name; technically it parses into a string pointer…

Comment (bsardo) [reply openrtb_ext/request_wrapper.go]: @scr-oath we discussed offline and are good with `gjson`. We discussed alternatives like using `jsonparser` which is used extensively throughout the code base but we could eventually move away from or changing the type of `gpc` to some custom type like `util/jsonutil/StringInt` which has an unmarshal method override.

I like that you moved this to a jsonutil function.@scr-oath we discussed offline and are good with `gjson`. We discussed alternatives like using `jsonparser` which is used extensively throughout the code base but we could eventually move away from or changing the type of `gpc` to some custom type like `util/jsonutil/StringInt` which has an unmarshal method override.

I like that you moved this to a jsonutil function.

File: util/jsonutil/forcestring.go
Comment (scr-oath) [reply util/jsonutil/forcestring.go]: DONE

File: openrtb_ext/request_wrapper.go
Comment (scr-oath) [reply openrtb_ext/request_wrapper.go]: gjson is actually quite fast… we use it a lot for the json.RawMessage stuff because just looking for a value is much faster than unmarshalling.

What I also love about it is that you can read json and then interrogate its structure.  We actually use it in our slog to OpenTelemetry Handler to convert into rich, recursive otel types (primitive, map, array) that correspond to the json counterparts.

**Review by bsardo — DISMISSED** [review]

**Comment by bsardo** [comment]
@scr-oath can you merge with master?

**Comment by scr-oath** [comment]
> @scr-oath can you merge with master?

Well it had conflicts by the time I got to it, so I need approvals again… I don't see any merge button at the moment though; do I need to be added to some github group to be able to merge after approval?

**Review by SyntaxNode — APPROVED** [review]

**Pull request merged**
